plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/QueryBuilder.java

```diff
@@ -170,6 +160,19 @@ protected String getProjection(List<JdbcColumnHandle> columns, Map<String, Strin
                 .collect(joining(", "));
     }
 
+    private String getFrom(JdbcRelationHandle baseRelation, Consumer<QueryParameter> accumulator)
+    {
+        if (baseRelation instanceof JdbcNamedRelationHandle) {
+            return " FROM " + getRelation(((JdbcNamedRelationHandle) baseRelation).getRemoteTableName());
+        }
+        if (baseRelation instanceof JdbcQueryRelationHandle) {
+            PreparedQuery preparedQuery = ((JdbcQueryRelationHandle) baseRelation).getPreparedQuery();
+            preparedQuery.getParameters().forEach(accumulator);
+            return " FROM (" + preparedQuery.getQuery() + ") o";
```

Comment:
nit: Since we know we would be passing a `List<QueryParameter>` can we pass a `Consumer<List<QueryParateter>>`. WDYT ?

Reply:
I've addressed https://github.com/trinodb/trino/pull/6892#discussion_r576007032
And I think it does not really matter.

Reply:
Oops I didn't see it.